csrc/dynamic_transform.cpp

```diff
@@ -807,17 +821,16 @@ bool DynamicTransformConcretizer::propagateFromProducerToConsumer(
 
     std::optional<IterType> id_type;
 
-    for (auto producer : ir_utils::filterByType<TensorView>(def->inputs())) {
-      PairwiseRootDomainMap root_map(producer, consumer);
-      auto c2p = root_map.mapConsumerToProducer(
-          consumer->domain(), producer->domain());
-
-      NVF_ERROR(
-          c2p.find(root_id) != c2p.end(),
-          "No input ID found to map with output ID: ",
-          root_id->toString());
+    for (const auto& c2p : c2p_maps) {
+      auto p_it = c2p.find(root_id);
+      if (p_it == c2p.end()) {
+        // We might not have mapped this ID to this producer, for example if the
```

Comment:
`PairwiseRootDomainMap` by default maps producer broadcast domains with consumer non-broadcast domains, so I think this comment is not correct.

Reply:
You're right. I think we should just assert `p_it != c2p.end()` like we were before.